src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -409,7 +409,9 @@ partition_downloader::download_log_with_capped_time(
     vlog(
       _ctxlog.info,
       "Starting log download with time limit of {}s",
-      retention_time.count() / 1000);
+      std::chrono::duration_cast<std::chrono::milliseconds>(retention_time)
+          .count()
+        / 1000);
```

Comment:
I wonder if this could be:
```suggestion
      "Starting log download with time limit of {}",
      std::chrono::duration_cast<std::chrono::seconds>(retention_time));
```